src/Compilers/Core/CodeAnalysisTest/Analyzers/AnalyzerFileReferenceAppDomainTests.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+#if NET472 // AppDomains
+
+using System;
+using System.Collections.Immutable;
+using System.Reflection;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Roslyn.Test.Utilities.Desktop;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.UnitTests
+{
+    public class RemoteAnalyzerFileReferenceTest : MarshalByRefObject
+    {
+        private Exception _analyzerLoadException;
+
+        public override object InitializeLifetimeService()
+        {
+            return null;
+        }
+
+        public Exception LoadAnalyzer(string analyzerPath)
+        {
+            _analyzerLoadException = null;
+            var analyzerRef = new AnalyzerFileReference(analyzerPath, FromFileLoader.Instance);
+            analyzerRef.AnalyzerLoadFailed += (s, e) => _analyzerLoadException = e.Exception;
+            var builder = ImmutableArray.CreateBuilder<DiagnosticAnalyzer>();
+            analyzerRef.AddAnalyzers(builder, LanguageNames.CSharp);
+            return _analyzerLoadException;
```

Comment:
Why is `_analyzerLoadException` being stored as a field here? It's only ever used in this method

Reply:
IDK. Seems like it could be a local. Fixing.